analysis/Scripts/NaiveYieldPlot.py

```diff
@@ -52,66 +50,133 @@
 	
 
 # CREATES THE DATASET FOR GRAPHING 
-y_dataDict = {}
-pt_dataDict = {}
+G_y_dataDict = {}
+G_pt_dataDict = {}
+for year in years:
+	G_y_dataDict[year] = {
+		"x" : [],
+		"y" : [],
+		"ex" : [],
+		"ey" : []
+	}
+
+	for points in y_arr_str:
+		G_y_dataDict[year]['x'].append(0)
+		G_y_dataDict[year]['y'].append(0)
+		G_y_dataDict[year]['ex'].append(0)
+		G_y_dataDict[year]['ey'].append(0)
+
+	G_pt_dataDict[year] = {
+		"x" : [],
+		"y" : [],
+		"ex" : [],
+		"ey" : []
+	}
+
+B_y_dataDict = {}
+B_pt_dataDict = {}
 for year in years:
-	y_dataDict[year] = {
+	B_y_dataDict[year] = {
 		"x" : [],
 		"y" : [],
 		"ex" : [],
 		"ey" : []
 	}
 
 	for points in y_arr_str:
-		y_dataDict[year]['x'].append(0)
-		y_dataDict[year]['y'].append(0)
-		y_dataDict[year]['ex'].append(0)
-		y_dataDict[year]['ey'].append(0)
+		B_y_dataDict[year]['x'].append(0)
+		B_y_dataDict[year]['y'].append(0)
+		B_y_dataDict[year]['ex'].append(0)
+		B_y_dataDict[year]['ey'].append(0)
 
-	pt_dataDict[year] = {
+	B_pt_dataDict[year] = {
 		"x" : [],
 		"y" : [],
 		"ex" : [],
 		"ey" : []
 	}
+
 	for points in pt_arr_str: #initializes the dataset to zero everywhere
-		pt_dataDict[year]['x'].append(0) # => "x" : [0,0,0,0,0,0,0] as theres 7 pt bins
-		pt_dataDict[year]['y'].append(0)
-		pt_dataDict[year]['ex'].append(0)
-		pt_dataDict[year]['ey'].append(0)
+		G_pt_dataDict[year]['x'].append(0) # => "x" : [0,0,0,0,0,0,0] as theres 7 pt bins
+		G_pt_dataDict[year]['y'].append(0)
+		G_pt_dataDict[year]['ex'].append(0)
+		G_pt_dataDict[year]['ey'].append(0)
+
+	for points in pt_arr_str:
+		B_pt_dataDict[year]['x'].append(0)
+		B_pt_dataDict[year]['y'].append(0)
+		B_pt_dataDict[year]['ex'].append(0)
+		B_pt_dataDict[year]['ey'].append(0)
     
-	for pol in combinedDict[year]:
+	for pol in GaussCB_combinedDict[year]:
+		i = 0
+		for file in sorted(GaussCB_combinedDict[year][pol]):
+			if ("ybin" in file) and ("Xic" in file):
+				y = (float(file[:-5].split('_')[-1].split('-')[0])+float(file[:-5].split('_')[-1].split('-')[1]))/2
+				y_err = (float(file[:-5].split('_')[-1].split('-')[1])-float(file[:-5].split('_')[-1].split('-')[0]))/2
+				ratio = GaussCB_combinedDict[year][pol][file]["yield_val"]/GaussCB_combinedDict[year][pol]['Lc'+file[3:]]["yield_val"]
+				ratio_err = math.sqrt((GaussCB_combinedDict[year][pol][file]["yield_err"]/GaussCB_combinedDict[year][pol][file]["yield_val"])**2 + (GaussCB_combinedDict[year][pol]['Lc'+file[3:]]["yield_err"]/GaussCB_combinedDict[year][pol]['Lc'+file[3:]]["yield_val"])**2)
+
+				G_y_dataDict[year]["x"][i] = y
+				G_y_dataDict[year]["y"][i] += ratio
+				G_y_dataDict[year]["ex"][i] = y_err
+
+				if G_y_dataDict[year]["ey"][i] == 0:
```

Comment:
why check if it is equal to 0 when it is set equal to 0 above

Reply:
Because i is an iterator that increases value every loop iteration (see l.129) ?

Reply:
Yeah but the value of i isn't checked, it is the value of G_y_dataDict[year]["ey"][i] and these are all set equal above (see l. 53-67)

Reply:
The initial values are set to 0, but for every ybin 'point' an entry is made in those lines. Then, in this loop, the value of 'i' just corresponds to the index of such a ybin (which is looped over in l.113).

I don't think this is the prettiest way to construct such a loop - but It seems to work.